File: libraries/hlp/errmes.f
      SUBROUTINE hlp_ERRMES (J, MES)
*+
*  - - - - - - -
*   E R R M E S
*  - - - - - - -
*
*  Translate help system error code into message.
*
*  Given (argument):
*     J         i           status value
*
*  Returned (argument)
*     MES       c*(*)       message (may be up to 50 characters)
*
*  P.T.Wallace   Starlink   28 July 1992
*-

      IMPLICIT NONE

      INTEGER J
      CHARACTER*(*) MES



      IF (J.GT.0) THEN
         MES='HLP illegal status'
      ELSE IF (J.EQ.0) THEN
         MES='OK'
      ELSE IF (J.EQ.-1) THEN
         MES='Help system in wrong state'
      ELSE IF (J.EQ.-2) THEN
         MES='Help library error on OPEN'
      ELSE IF (J.EQ.-3) THEN
         MES='Help library error on WRITE'
      ELSE IF (J.EQ.-4) THEN
         MES='Help library error on READ'
      ELSE IF (J.EQ.-5) THEN
         MES='Help library error on CLOSE'
      ELSE IF (J.EQ.-6) THEN
         MES='Attempt to WRITE outside help library'
      ELSE IF (J.EQ.-7) THEN
         MES='Attempt to READ outside help library'
      ELSE IF (J.EQ.-8) THEN
         MES='Help record overflows supplied string'
      ELSE IF (J.EQ.-9) THEN
         MES='Help library creation failure'
      ELSE IF (J.EQ.-10) THEN
         MES='HLP illegal status'
      ELSE IF (J.EQ.-11) THEN
         MES='hlp_HELP internal error'
      ELSE IF (J.EQ.-12) THEN
         MES='Line output failure'
      ELSE IF (J.EQ.-13) THEN
         MES='Line input failure'
      ELSE IF (J.EQ.-14) THEN
         MES='Invalid index entry'
      ELSE IF (J.EQ.-15) THEN
         MES='Attempted switch to current library'
      ELSE IF (J.EQ.-16) THEN
         MES='String too small'
      ELSE IF (J.EQ.-17) THEN
         MES='File name translation error'
      ELSE
         MES='HLP illegal status'
      END IF

      END

File: libraries/hlp/errmes.c
#include "help.h"
#include "hlpsys.h"

char* hlpErrmes ( int nerror )
/*
**  - - - - - - - - - -
**   h l p E r r m e s
**  - - - - - - - - - -
**
**  Translate HELP system error code into message.
**
**  Given (argument):
**     nerror    int         error code
**
**  Returned (function value)
**               char*       message (may be up to 50 characters)
**
**  The HELP system error codes are defined in the hlpsys.h header
**  file.
**
**  Last revision:   14 January 2008
**
**  Copyright P.T.Wallace.  All rights reserved.
*/
{
   switch ( nerror ) {
   case 0:
      return "HLP: OK";
   case hlp_ILLEGAL_STATE:
      return "HLP: help system in wrong state";
   case hlp_OPEN_ERROR:
      return "HLP: help library error on open";
   case hlp_WRITE_ERROR:
      return "HLP: help library error on write";
   case hlp_READ_ERROR:
      return "HLP: help library error on read";
   case hlp_CLOSE_ERROR:
      return "HLP: help library error on close";
   case hlp_WRITE_WIDE:
      return "HLP: attempt to write outside help library";
   case hlp_READ_WIDE:
      return "HLP: attempt to read outside help library";
   case hlp_RECORD_OVERSIZE:
      return "HLP: help record overflows supplied string";
   case hlp_CREATION_FAILURE:
      return "HLP: help library creation failure";
   case hlp_INTERNAL_ERROR:
      return "HLP: help illegal status";
   case hlp_ILLEGAL_LEVEL:
      return "HLP: illegal current level";
   case hlp_LINE_OUTPUT_BAD:
      return "HLP: line output failure";
   case hlp_LINE_INPUT_BAD:
      return "HLP: line input failure";
   case hlp_BAD_INDEX:
      return "HLP: invalid index entry";
   case hlp_SELF_REF:
      return "HLP: attempted switch to current library";
   case hlp_STRING_OVERFLOW:
      return "HLP: string too small";
   case hlp_TRANSLATE_ERROR:
      return "HLP: file name translation error";
   default:
      return "HLP: illegal status";
   }
}
